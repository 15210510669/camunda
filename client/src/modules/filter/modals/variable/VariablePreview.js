/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React from 'react';

import {t} from 'translation';

import PreviewItemValue from '../../PreviewItemValue';

export default function VariablePreview({variableName, filter: {operator, values}}) {
  const createOperator = (val) => <span> {val} </span>;

  const operatorText = (
    <span>
      {operator === 'not in' || operator === 'not contains'
        ? createOperator(t('common.filter.list.operators.nor'))
        : createOperator(t('common.filter.list.operators.or'))}
    </span>
  );

  return (
    <>
      <span className="parameterName">{variableName}</span>
      {(!operator || operator === 'in' || operator === '=') &&
        createOperator(t('common.filter.list.operators.is'))}
      {operator === 'contains' && createOperator(t('common.filter.list.operators.contains'))}
      {operator === 'not contains' && createOperator(t('common.filter.list.operators.notContains'))}
      {operator === 'not in' &&
        (values.length === 1
          ? createOperator(t('common.filter.list.operators.not'))
          : createOperator(t('common.filter.list.operators.neither')))}
      {operator === '<' && createOperator(t('common.filter.list.operators.less'))}
      {operator === '>' && createOperator(t('common.filter.list.operators.greater'))}
      {values.map((value, idx) => (
        <span key={idx}>
          {value === null ? (
            <>
              <PreviewItemValue>{t('common.null')}</PreviewItemValue>
              {operatorText}
              <PreviewItemValue>{t('common.undefined')}</PreviewItemValue>
            </>
          ) : (
            <PreviewItemValue>{value.toString()}</PreviewItemValue>
          )}
          {idx < values.length - 1 && operatorText}
        </span>
      ))}
    </>
  );
}
