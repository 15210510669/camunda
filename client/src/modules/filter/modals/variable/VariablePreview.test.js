/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React from 'react';
import {mount} from 'enzyme';

import VariablePreview from './VariablePreview';

it('should combine multiple variable values with or', () => {
  const node = mount(
    <VariablePreview
      filter={{operator: 'in', values: ['varValue', 'varValue2']}}
      variableName="varName"
    />
  );

  expect(node).toIncludeText('varName is varValue or varValue2');
});

it('should combine multiple variable values with or', () => {
  const node = mount(
    <VariablePreview
      filter={{operator: 'not in', values: [null, 'varValue', 'varValue2']}}
      variableName="varName"
    />
  );

  expect(node).toIncludeText('varName is neither null nor undefined nor varValue nor varValue2');
});

it('should use less/greater for comparison operators', () => {
  const node = mount(
    <VariablePreview filter={{operator: '<', values: ['varValue']}} variableName="varName" />
  );

  expect(node).toIncludeText('varName is less than varValue');
});

it('should use contains operators', () => {
  const node = mount(
    <VariablePreview filter={{operator: 'contains', values: ['varValue']}} variableName="varName" />
  );

  expect(node).toIncludeText('varName contains varValue');
});

it('should display correct preview even if no operator is defined', () => {
  const node = mount(<VariablePreview filter={{values: [true, false]}} variableName="varName" />);

  expect(node).toIncludeText('varName is true or false');
});
