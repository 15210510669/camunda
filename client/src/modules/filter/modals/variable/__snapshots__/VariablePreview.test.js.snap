// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should combine multiple variable values with or 1`] = `
<Fragment>
  <span
    className="parameterName"
  >
    varName
  </span>
  <span>
     
    is
     
  </span>
  <span
    key="0"
  >
    <PreviewItemValue>
      varValue
    </PreviewItemValue>
    <span>
      <span>
         
        or
         
      </span>
    </span>
  </span>
  <span
    key="1"
  >
    <PreviewItemValue>
      varValue2
    </PreviewItemValue>
  </span>
</Fragment>
`;

exports[`should combine multiple variable values with or 2`] = `
<Fragment>
  <span
    className="parameterName"
  >
    varName
  </span>
  <span>
     
    is neither
     
  </span>
  <span
    key="0"
  >
    <PreviewItemValue>
      null
    </PreviewItemValue>
    <span>
      <span>
         
        nor
         
      </span>
    </span>
    <PreviewItemValue>
      undefined
    </PreviewItemValue>
    <span>
      <span>
         
        nor
         
      </span>
    </span>
  </span>
  <span
    key="1"
  >
    <PreviewItemValue>
      varValue
    </PreviewItemValue>
    <span>
      <span>
         
        nor
         
      </span>
    </span>
  </span>
  <span
    key="2"
  >
    <PreviewItemValue>
      varValue2
    </PreviewItemValue>
  </span>
</Fragment>
`;

exports[`should display correct preview even if no operator is defined 1`] = `
<Fragment>
  <span
    className="parameterName"
  >
    varName
  </span>
  <span>
     
    is
     
  </span>
  <span
    key="0"
  >
    <PreviewItemValue>
      true
    </PreviewItemValue>
    <span>
      <span>
         
        or
         
      </span>
    </span>
  </span>
  <span
    key="1"
  >
    <PreviewItemValue>
      false
    </PreviewItemValue>
  </span>
</Fragment>
`;

exports[`should use contains operators 1`] = `
<Fragment>
  <span
    className="parameterName"
  >
    varName
  </span>
  <span>
     
    contains
     
  </span>
  <span
    key="0"
  >
    <PreviewItemValue>
      varValue
    </PreviewItemValue>
  </span>
</Fragment>
`;

exports[`should use less/greater for comparison operators 1`] = `
<Fragment>
  <span
    className="parameterName"
  >
    varName
  </span>
  <span>
     
    is less than
     
  </span>
  <span
    key="0"
  >
    <PreviewItemValue>
      varValue
    </PreviewItemValue>
  </span>
</Fragment>
`;
